### ISSUE TITLE
HTTP Statuscode and overall status wrong when a service is unhealthy 

### ISSUE THREAD

**Issue by 0xC0DEBA5E** [matteocacciola/fastapi_healthz#1]
Problem:

 When a service is unhealthy, the HTTP Statuscode is not set to 500. It is 200 instead. Also, the `status` is `HealthCheckStatusEnum.UNHEALTHY`. 

Even when all services are healthy, the status is `HealthCheckStatusEnum.HEALTHY` which I think is also not intended as it should be `HEALTHY` instead. 
 
 I suppose this comes from a wrong string conversion of the `HealthCheckStatusEnum`. We tried to fix it and it seemed to work when we added the code for the enum to string conversion in `model.py`:
 
Base version:
 ```python
 class HealthCheckStatusEnum(BaseEnum):
    HEALTHY = "Healthy"
    UNHEALTHY = "Unhealthy"
```
Modified version
```python
 class HealthCheckStatusEnum(BaseEnum):
    def __str__(self):
        return str(self.value)

    HEALTHY = "Healthy"
    UNHEALTHY = "Unhealthy"
```

This might not be an optimal solution and needs further testing, but at least it shows where the problem seems to be.

**Comment by matteocacciola** [comment]
>
> When a service is unhealthy, the HTTP Statuscode is not set to 500. It is
> 200 instead. Also, the status is HealthCheckStatusEnum.UNHEALTHY.
>
200 is intended and the returned status set to HealthCheckStatusEnum.
UNHEALTHY is correct.

> Even when all services are healthy, the status is
> HealthCheckStatusEnum.HEALTHY which I think is also not intended as it
> should be HEALTHY instead.
>
When all services are healthy, the status is HealthCheckStatusEnum.HEALTHY,
which is correct and intended, since an healthy status is... well...
healthy.

**Comment by 0xC0DEBA5E** [comment]
> 200 is intended and the returned status set to HealthCheckStatusEnum. UNHEALTHY is correct.

Sorry, maybe I explained my problem poorly. 

As per my understanding of the [Readme Section about returned-data](https://github.com/matteocacciola/fastapi_healthz?tab=readme-ov-file#returned-data) the status code should be 500.

If I make a request when everything is healthy, I get the following response with status code 200 and the following JSON Body:

```json
{"status":"HealthCheckStatusEnum.HEALTHY","total_time_taken":"0:00:00.000004","entities":[]}
```
I believe the intention was that the response body should instead look like this:

```json
{"status":"HEALTHY","total_time_taken":"0:00:00.000004","entities":[]}
```

When there is a failing health check (i.e. the db), the status code was 200 and the response body looked like this:
```json
{"status":"HealthCheckStatusEnum.UNHEALTHY","total_time_taken":"0:00:00.000982","entities":[{"service":"db","status":"Unhealthy","time_taken":"0:00:00.000961","tags":["db"]}]}
```
I think the status code should be 500 and the response body should look like this instead:
```json
{"status":"UNHEALTHY","total_time_taken":"0:00:00.000982","entities":[{"service":"db","status":"Unhealthy","time_taken":"0:00:00.000961","tags":["db"]}]}
```

I think the status code when a service is unhealthy should be 500, as you can see  [in route.py](https://github.com/matteocacciola/fastapi_healthz/blob/master/fastapi_healthz/route.py#L25C1-L29C71) .
The problem with this code is, that (at least on my machine where I was debugging it) the condition `if res["status"] == HealthCheckStatusEnum.UNHEALTHY:` never will be true, because we are comparing a str with an enum here. So the next line `return JSONResponse(content=encode_json(res), status_code=500)` will never be executed.

I hope this helps in clarifying what the problem is that I'm facing.

**Comment by matteocacciola** [comment]
Oh, now I see. Fixed by b8ce102abba3607e52076ff2d20f17dbcf8405ae
